Classes/EventListener/Order/Stock/HandleStock.php

```diff
@@ -11,77 +11,47 @@
     {
-        $productConnection = GeneralUtility::makeInstance(ConnectionPool::class)
-            ->getConnectionForTable('tx_cartproducts_domain_model_product_product');
-        $productQueryBuilder = $productConnection->createQueryBuilder();
-
-        $product = $productQueryBuilder
-            ->select('stock')
-            ->from('tx_cartproducts_domain_model_product_product')->where($productQueryBuilder->expr()->eq('uid', $productQueryBuilder->createNamedParameter($cartProduct->getProductId(), \PDO::PARAM_INT)))->executeQuery()->fetchAssociative();
-
-        $productQueryBuilder
-            ->update('tx_cartproducts_domain_model_product_product')
-            ->where(
-                $productQueryBuilder->expr()->eq('uid', $productQueryBuilder->createNamedParameter($cartProduct->getProductId(), \PDO::PARAM_INT))
-            )
-            ->orWhere(
-                $productQueryBuilder->expr()->eq('l10n_parent', $productQueryBuilder->createNamedParameter($cartProduct->getProductId(), \PDO::PARAM_INT))
-            )->set('stock', $product['stock'] - $cartProduct->getQuantity())->executeStatement();
+        $quantityToAdd = -1 * $cartProduct->getQuantity();
+        $this->productRepository->addQuantityToStock($cartProduct->getProductId(), $quantityToAdd);
     }
 
-    protected function handleStockInBeVariant(CartProduct $cartProduct): void
+    private function handleStockInBeVariant(CartProductBeVariant $cartProductBeVariant): void
     {
-        $beVariantConnection = GeneralUtility::makeInstance(ConnectionPool::class)
-            ->getConnectionForTable('tx_cartproducts_domain_model_product_bevariant');
-
-        foreach ($cartProduct->getBeVariants() as $cartBeVariant) {
-            $beVariantQueryBuilder = $beVariantConnection->createQueryBuilder();
-            $beVariant = $beVariantQueryBuilder
-                ->select('stock')
-                ->from('tx_cartproducts_domain_model_product_bevariant')->where($beVariantQueryBuilder->expr()->eq('uid', $beVariantQueryBuilder->createNamedParameter($cartBeVariant->getId(), \PDO::PARAM_INT)))->executeQuery()->fetchAssociative();
-
-            $beVariantQueryBuilder
-                ->update('tx_cartproducts_domain_model_product_bevariant')
-                ->where(
-                    $beVariantQueryBuilder->expr()->eq('uid', $beVariantQueryBuilder->createNamedParameter($cartBeVariant->getId(), \PDO::PARAM_INT))
-                )
-                ->orWhere(
-                    $beVariantQueryBuilder->expr()->eq('l10n_parent', $beVariantQueryBuilder->createNamedParameter($cartBeVariant->getId(), \PDO::PARAM_INT))
-                )->set('stock', $beVariant['stock'] - $cartBeVariant->getQuantity())->executeStatement();
-        }
+        $quantityToAdd = -1 * $cartProductBeVariant->getQuantity();
+        $this->beVariantRepository->addQuantityToStock((int)$cartProductBeVariant->getId(), $quantityToAdd);
```

Comment:
Same as above.

Reply:
I added an method to the repository in 9907132.